main.cpp

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
```

Comment:
first bucket should accomodate one more element because 1000000 will be coming in at 1000th index

Reply:
So add a condition for that case